ethcore/src/client/private_notify.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use error::TransactionImportError;
+
+/// Represent private transactions handler inside the client
+pub trait PrivateNotify : Send + Sync {
+	/// fires when private transaction message queued via client io queue
+	fn private_transaction_queued(&self) -> Result<(), TransactionImportError> {
+		// does nothing by default
+		Ok(())
```

Comment:
this trait has only one function. I believe it's not a good pattern to have a default implementation for the only one function